src/lib/utils.test.ts

```diff
@@ -32,3 +38,1209 @@ describe('testing roundOff', () => {
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '1.66'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with 0.375', () => {
+        const match = simulateOnMouseDragMatch('0.375')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '1.375'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with 0.1', () => {
+        const match = simulateOnMouseDragMatch('0.1')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '1.1'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with .0', () => {
+        const match = simulateOnMouseDragMatch('.0')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '1.0'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with 1.0', () => {
+        const match = simulateOnMouseDragMatch('1.0')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '2.0'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with 1.66', () => {
+        const match = simulateOnMouseDragMatch('1.66')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '2.66'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with 0.0000001', () => {
+        const match = simulateOnMouseDragMatch('0.0000001')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '1.0000001'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with 9', () => {
+        const match = simulateOnMouseDragMatch('9')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '10'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with 99', () => {
+        const match = simulateOnMouseDragMatch('99')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '100'
+        expect(actual).toBe(expected)
+      })
+
+      test('works with .123', () => {
+        const match = simulateOnMouseDragMatch('.123')
+        const matchedText = match ? match[0] : ''
+        const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
+        const expected = '1.123'
+        expect(actual).toBe(expected)
+      })
```

Comment:
We sometimes pull out the literals of similar tests into an object, and instead run the tests by mapping over the array of objects. I think it up for debate if that's more readable than what you've done here, but what do you think about that? I think doing this sort of thing too many levels deep makes things less readable, but maybe just one level could help.

```suggestion
      const onesEventCases = [
        {
          startValue: '0.0',
          endValue: '1.0'
        },
        {
          startValue: '1',
          endValue: '2'
        },
        {
          startValue: '0.5',
          endValue: '1.5'
        },
        {
          startValue: '66',
          endValue: '1.66'
        },
        {
          startValue: '0.375',
          endValue: '1.375'
        },
        {
          startValue: '0.1',
          endValue: '1.1'
        },
        {
          startValue: '.0',
          endValue: '1.0'
        },
        {
          startValue: '1.0',
          endValue: '2.0'
        },
        {
          startValue: '1.66',
          endValue: '2.66'
        },
        {
          startValue: '0.0000001',
          endValue: '1.0000001'
        },
        {
          startValue: '9',
          endValue: '10'
        },
        {
          startValue: '99',
          endValue: '100'
        },
        {
          startValue: '.123',
          endValue: '1.123'
        }
      ]
      
      onesEventCases.map(({ startValue, endValue }) => {
        test(`works with ${startValue}`, () => {
          const match = simulateOnMouseDragMatch(startValue)
          const matchedText = match ? match[0] : ''
          const actual = onDragNumberCalculation(matchedText, positiveOnesEvent)
          const expected = endValue
          expect(actual).toBe(expected)
        })
      })
```